moved/src/state_actor.rs

```diff
@@ -171,6 +171,16 @@ impl<
                 self.mem_pool
                     .insert(tx_hash, (ExtendedTxEnvelope::Canonical(tx), encoded));
             }
+            StateMessage::GetBalance {
+                address,
+                response_channel,
+                ..
+            } => {
+                // TODO: Support balance from arbitrary blocks
```

Comment:
This should be doable via the merkle tree. Look up the block based on the number/tag, get the state root from that block and then use the merkle store to construct a state instance corresponding to that root.

Reply:
Created an issue to tackle this independently: https://github.com/MovedNetwork/op-move/issues/142